### PR TITLE
Fix repair beams not deallocating

### PR DESCRIPTION
**Pull request by ohemelaar**
The solution was present in the harvester code

### CODE REVIEW

**Comment by ThibaultLemaire** [comment]
Yep looks like it does the trick.

File: src/Ships/RepairCorvette.c [hunk 1]
```diff
@@ -221,7 +221,7 @@ void stopRepairEffect(Ship *ship)
 
     if (bitTest(((etgeffectstatic *)ship->rceffect->staticinfo)->specialOps, ESO_SelfDeleting))
     {                                                       //if the effect will delete itself
-        ((real32 *)ship->rceffect->variable)[ETG_ResourceDurationParam] =
+        ((real32 *)ship->rceffect->variable)[SCALESHIFT(ETG_ResourceDurationParam)] =
```
Comment (ThibaultLemaire) [inline src/Ships/RepairCorvette.c]: `SCALECAST` I have seen it used in relation to 64b code. However, I have no clue what it's accomplishing here.

Comment (ThibaultLemaire) [reply src/Ships/RepairCorvette.c]: Oh wait I think I get it. real32 is still 4 bytes of size, but we're passing 8 bytes arguments to ETG functions. So when we're writing in the parameter array, we have to jump twice the size of a real32 before getting to the beginning of the next parameter.

Comment (ThibaultLemaire) [reply src/Ships/RepairCorvette.c]: But then you should also update the ETG_ResourceLengthParam and ETG_ResourceRadiusParam (the positions of the other params)

Actually, why not apply the scale shift at their definition ?

Comment (ohemelaar) [reply src/Ships/RepairCorvette.c]: I have to confess that I didn't really know _why_ this fixed the issue. I'll check it again using you insights tomorrow.

**Comment by ohemelaar** [comment]
I've looked for the usages of the other defines that appear in ETG.h and they don't always appear to be used with SCALESHIFT. Some more ETG bugs may be caused by this.

**Review by ThibaultLemaire — CHANGES_REQUESTED** [review]
Yes, well for the first parameter, multiplying 0 by 2 still makes it 0, but the second parameter should be `SCALESHIFT`ed as well. It's the logical thing to do (though I don't know why it hasn't caused any bugs yet)

Or maybe you think this should go to an other PR?

**Comment by ohemelaar** [comment]
Yes I think it should be in another PR as this one fixes the bug, and we don't know whether the other variables are bugged or not

**Pull request merged**